### ISSUE TITLE
ecds: add support for access log extensions

### ISSUE THREAD

**Issue by kyessenov** [envoyproxy/envoy#12893]
This is a tracking issue to implement ECDS for access log extensions. This can be used to dynamically update access log format without reloading HCM/listener, for example.

**Comment by jpeach** [comment]
@kyessenov Are there any more details about `ECDS`?

**Comment by kyessenov** [comment]
https://www.envoyproxy.io/docs/envoy/latest/configuration/overview/extension#discovery-service. Right now, it's just HCM that supports it.

**Comment by zirain** [comment]
@kyessenov are you still working on this? I think we need this for #21718

**Comment by kyessenov** [comment]
@zirain No, I'm not. I agree this is useful for fine-grained access log configuration.
